support-frontend/assets/pages/contributions-checkout-in-epic/ContributionsCheckout.tsx

```diff
@@ -0,0 +1,171 @@
+	const amounts = state.common.amounts;
+
+	return {
+		ONE_OFF: {
+			...amounts.ONE_OFF,
+			amounts: amounts.ONE_OFF.amounts.slice(0, 2),
+		},
+		MONTHLY: {
+			...amounts.MONTHLY,
+			amounts: amounts.MONTHLY.amounts.slice(0, 2),
+		},
+		ANNUAL: {
+			...amounts.ANNUAL,
+			amounts: amounts.ANNUAL.amounts.slice(0, 2),
+		},
+	};
+};
+
+const mapStateToProps = (state: State) => ({
+	status: getStatus(state),
+	country: state.common.internationalisation.countryId,
+	countryGroupId: state.common.internationalisation.countryGroupId,
+	currency: state.common.internationalisation.currencyId,
+	contributionType: state.page.form.contributionType,
+	amounts: getAmounts(state),
+	selectedAmounts: state.page.form.selectedAmounts,
+	otherAmounts: state.page.form.formData.otherAmounts,
+});
+
+const mapDispatchToProps = (dispatch: ThunkDispatch<State, void, Action>) => {
+	return {
+		setSelectedContributionType: (contributionType: ContributionType) => {
+			dispatch(
+				updateContributionTypeAndPaymentMethod(contributionType, 'None'),
+			);
+		},
+		setSelectedAmount: (
+			amount: number | 'other',
+			contributionType: ContributionType,
+		) => {
+			dispatch(selectAmount(amount, contributionType));
+		},
+		setOtherAmount: (amount: string, contributionType: ContributionType) => {
+			dispatch(updateOtherAmount(amount, contributionType));
+		},
+	};
+};
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+
+type ContributionsCheckoutProps = ConnectedProps<typeof connector>;
+
+function ContributionsCheckout({
+	status,
+	country,
+	countryGroupId,
+	currency,
+	contributionType,
+	amounts,
+	selectedAmounts,
+	otherAmounts,
+	setSelectedContributionType,
+	setSelectedAmount,
+	setOtherAmount,
+}: ContributionsCheckoutProps): JSX.Element {
+	const ref = useOnHeightChangeEffect((height) => {
+		window.parent.postMessage({ type: 'IFRAME_HEIGHT', height }, '*');
+	});
+
+	const secondaryCta = useSecondaryCta();
+
+	return (
+		<div ref={ref} css={styles.container}>
+			{status === 'FORM' && (
+				<ContributionsCheckoutForm
+					country={country}
+					countryGroupId={countryGroupId}
+					currency={currency}
+					contributionType={contributionType}
+					amounts={amounts}
+					selectedAmounts={selectedAmounts}
+					otherAmounts={otherAmounts}
+					setSelectedContributionType={setSelectedContributionType}
+					setSelectedAmount={(amount) =>
+						setSelectedAmount(amount, contributionType)
+					}
+					setOtherAmount={(amount) => setOtherAmount(amount, contributionType)}
+					secondaryCta={secondaryCta}
+				/>
+			)}
+
+			{status === 'SUBMITTING' && <ContributionsCheckoutSubmitting />}
+
+			{status === 'SUCCESS' && (
+				<ContributionsCheckoutThankYou
+					currency={currency}
+					contributionType={contributionType}
+					amount={getAmount(selectedAmounts, otherAmounts, contributionType)}
+				/>
+			)}
+		</div>
+	);
+}
+
+// ---- Hooks ---- //
+
+function useOnHeightChangeEffect(onChange: (height: number) => void) {
```

Comment:
I think it would be good to move this into `assets/helpers/customHooks` rather than having it inline in this file- makes it more discoverable if we might want to reuse it elsewhere.

Reply:
Yeah great suggestion - I've moved it over and added just a little comment to describe what it does